src/components/grouped-character/grouped-character.test.tsx

```diff
@@ -0,0 +1,451 @@
+  const onBlur = jest.fn();
+  const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+
+  render(
+    <MockComponent
+      initialValue="12345678"
+      onBlur={(ev) => {
+        onBlur?.(ev.target.value);
+      }}
+    />
+  );
+
+  const input = screen.getByRole("textbox");
+
+  await user.click(input);
+  await user.clear(input);
+  await user.type(screen.getByRole("textbox"), "123456");
+  await user.tab();
+
+  expect(onBlur).toHaveBeenLastCalledWith(
+    expect.objectContaining({
+      formattedValue: "12-34-56",
+      rawValue: "123456",
+    })
+  );
+});
+
+// coverage
+// This test scenario may seem obvious but it is required for coverage
+test("does nothing if onBlur is not provided", async () => {
+  const onChange = jest.fn();
+  const onBlur = jest.fn();
+  const user = userEvent.setup({ delay: null });
+
+  render(
+    <GroupedCharacter
+      separator="-"
+      value="12345678"
+      groups={[2, 2, 4]}
+      onChange={onChange}
+      onBlur={undefined}
+    />
+  );
+
+  const input = screen.getByRole("textbox");
+
+  await user.click(input);
+  await user.clear(input);
+  await user.type(screen.getByRole("textbox"), "123456");
+  await user.tab();
+
+  expect(onBlur.mock.calls[0]).toBe(undefined);
+});
+
+test("does not allow values of length greater than that allowed by the group config", () => {
+  render(
+    <GroupedCharacter separator="-" value="1234567890" groups={[2, 2, 4]} />
+  );
+
+  expect(screen.getByRole("textbox")).toHaveValue("12-34-5678");
+});
+
+test("does not allow any more characters to be entered when value string is at max length", async () => {
+  const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+
+  render(<MockComponent initialValue="12-34-5678" />);
+
+  const input = screen.getByRole("textbox");
+
+  await user.click(input);
+  await user.clear(input);
+  await user.type(input, "123456789");
+
+  expect(input).toHaveValue("12-34-5678");
+});
+
+test("pressing backspace after a separating character moves cursor the backwards, if the cursor is positioned at the end of input", async () => {
+  const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+
+  render(<MockComponent initialValue="" />);
+
+  const input = screen.getByRole("textbox") as HTMLInputElement;
+
+  await user.click(input);
+  await user.clear(input);
+  await user.type(input, "12-34-");
+  input.setSelectionRange(6, 6);
+  await user.keyboard("{backspace}");
+
+  expect(input.selectionStart).toBe(4);
+  expect(input.selectionEnd).toBe(4);
+
+  await user.click(input);
+  await user.clear(input);
+  await user.type(input, "12-");
+  input.setSelectionRange(3, 3);
+  await user.keyboard("{backspace}");
+
+  expect(input.selectionStart).toBe(1);
+  expect(input.selectionEnd).toBe(1);
+});
+
+test("placing cursor at the point where a separator should appear, it moves the cursor forward one position", async () => {
```

Comment:
**suggestion** (non-blocking): it's really not clear what "it" means here without reading the test. (Reading the test, "it" appears to be "pressing tab", which confuses me as I mentioned [in a comment on my last review](https://github.com/Sage/carbon/pull/6897#discussion_r1723386541) - but so be it I guess. The original description was "pressing a character", which to me means a letter or number, I'd rather change the description to something like that if we can change the test so that it's doing that rather than tab.)